File: mainFolder/Location.java
import java.util.*;

public class Location {
    private String name;
    private String description;
    private ArrayList<Adventurer> adventurers;

    public Location(String name, String description) {
        this.name = name;
        this.description = description;
        this.adventurers = new ArrayList<>();
    }

    public String getName() {
        return name; // gets name
    }

    public ArrayList<Adventurer> getAdventurers() {
        return adventurers; // gets all adventurers
    }

    public String getDescription() {
        return description; // returns description
    }

    public void addAdventurer(Adventurer adventurer) {
        this.adventurers.add(adventurer); // adds an adventurer to arraylist adventurers
    }

    public void removeAdventurer(Adventurer adventurer) {
        this.adventurers.remove(adventurer); // removes an adventurer from arraylist adventurers
    }
}

File: mainFolder/location.rs
use crate::adventurer::Adventurer;

#[derive(Clone)]
pub struct Location {
    pub name:        String,
    pub description: String,
    pub adventurers: Vec<Adventurer>,
}

impl Location {
    pub fn new(name: String, description: String) -> Self {
        Location {
            name,
            description,
            adventurers: Vec::new(),
        }
    }

    pub fn get_name(&self) -> &str {
        &self.name
    }

    pub fn get_description(&self) -> &str {
        &self.description
    }

    pub fn add_adventurer(&mut self, adventurer: Adventurer) {
        self.adventurers.push(adventurer);
    }

    pub fn remove_adventurer(&mut self, adventurer: &Adventurer) {
        self.adventurers.retain(|adv| adv.get_name() != adventurer.get_name());
    }
}